lib/libtsunami/src/animation.cpp

```diff
@@ -29,21 +29,33 @@ void Animation::triggerAdd(Trigger *t) {
 }
 
 void Animation::triggerRemove(Trigger *tr) {
-	LockVideo();
+	int lockedState = 0;
+	if (!(lockedState = VideoIsLocked())) {
```

Comment:
Seems you have recursive locking. This is bad solution, better make video mutex as recursive at least.

Reply:
In the animation and triggers I must leave them, because the libraries natively delete the objects, in the rest I put it just to provide a solution.

Reply:
To solve what? This condition may not work, because IRQ may occur in the interval between the calls of these functions and the comparison of the result.

Reply:
In the animations and triggers, they are deleted in the video thread natively, it is not something that I put.
When I delete objects I could be deleting animations.

Reply:
I already did the rollback, just from that.

Reply:
Do you mean in the title of the PR?

Reply:
animation, drawable, banner cpp files: https://github.com/DC-SWAT/DreamShell/pull/77/files

Reply:
But they are already deleted with the new commit.
I don't understand what you mean.